### ISSUE TITLE
anim.begin callback never called if anim.delay is a function

### ISSUE THREAD

**Issue by singuerinc** [juliangarnier/anime#79]
Since `begin` is executed when `time.current >= s.delay` then if `s.delay` is a function the statement always return false.

https://github.com/juliangarnier/anime/blob/master/anime.js#L550

I'm trying to animate multiple objects and return a delay based on the index:

```
anime({
  targets: "ul li",
  begin: (animation) => console.log("begin"),
  opacity: [0, 1],
  duration: 1500,
  delay: (el, index) => 25 * index
});
```

Check out this test I have made:
http://codepen.io/singuerinc/pen/jrBVdJ

My example is not complicated to solve since is using the index, but I'm not sure how to solve it, for instance, if you use a more complex calculation.

**Comment by alexchantastic** [comment]
I am seeing the same issue. However, if it is a named function, it is called. For example,

```
function foo() {
     // stuff here
};

anime({
    targets: '#foo',
    begin: foo()
});
```

**Comment by singuerinc** [comment]
@alexchantastic that's incorrect, is not anime that is calling the function, is you when you write `foo()`.
You should write just `begin: foo` without `()`

**Comment by alexchantastic** [comment]
@singuerinc yes, you're correct, but I think `foo` will still be executed at the correct time. This is just a workaround. I suppose it's no different than:

```
foo();
anime({...});
```

**Comment by juliangarnier** [comment]
Will look at this later, thanks for pointing this out!

**Comment by alexchantastic** [comment]
@juliangarnier I've proposed a fix for this in PR #88.

**Comment by juliangarnier** [comment]
fixed in v1.1.2
